src/argilla/client/feedback/metrics/annotator_metrics.py

```diff
@@ -0,0 +1,251 @@
+#  Copyright 2021-present, the Recognai S.L. team.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+"""This module contains metrics to compare Annotator's suggestions vs responses. """
+
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from typing import TYPE_CHECKING, Any, Dict, List, Literal, Tuple, Union
+
+import numpy as np
+from pydantic import BaseModel
+
+from argilla.client.feedback.schemas import (
+    LabelQuestion,
+    MultiLabelQuestion,
+    RankingQuestion,
+    RatingQuestion,
+    TextQuestion,
+)
+from argilla.utils.dependency import requires_dependencies
+
+if TYPE_CHECKING:
+    from argilla.client.feedback.dataset import FeedbackDataset
+
+import evaluate
+import pandas as pd
+from sklearn.metrics import (
+    accuracy_score,
+    confusion_matrix,
+    f1_score,
+    multilabel_confusion_matrix,
+    precision_score,
+    recall_score,
+)
+
+# Type aliases
+Responses = List[Union[float, int, str]]
+Suggestions = Responses
+
+
+class AnnotatorMetric:
+    def __init__(self, dataset: "FeedbackDataset", question_name: str) -> None:
+        self._dataset = dataset
+        self._question_name = question_name
+        self._question_type = type(self._dataset.question_by_name(question_name))
+        if self._question_type in (MultiLabelQuestion, RankingQuestion):
+            raise NotImplementedError(f"No metrics are defined currently for {self._question_type.__name__}")
+        self._allowed_metrics = METRICS_PER_QUESTION[self._question_type]
+
+    def _prepare_responses_and_suggestions(self) -> Tuple[Dict[int, Responses], Suggestions]:
```

Comment:
@plaguss, we differentiate between responses with different statuses like draft, pending and submitted, we should keep this in mind because we only want to include submitted records IMO.

Reply:
The new function does it now (for the moment I'll keep it here, once the new `argilla.client.feedback.utils` module is ready I'll move it there as well as the tests)